modules/wallet/defrag.go

```diff
@@ -12,9 +12,17 @@ var (
 	errDefragNotNeeded = errors.New("defragging not needed, wallet is already sufficiently defragged")
 )
 
-// createDefragTransaction creates a transaction that spends multiple existing
+// managedCreateDefragTransaction creates a transaction that spends multiple existing
 // wallet outputs into a single new address.
-func (w *Wallet) createDefragTransaction() ([]types.Transaction, error) {
+func (w *Wallet) managedCreateDefragTransaction() ([]types.Transaction, error) {
+	// get the transaction fee
+	minFee, _ := w.tpool.FeeEstimation()
+	sizeAvgOutput := uint64(250)
+	fee := minFee.Mul64(sizeAvgOutput * defragBatchSize)
```

Comment:
Sorry to keep iterating on this, but this is another pattern that makes it easier to slip bugs into the code. The fee here is calculated completely separately from the place where the transactions are created, which means verifying correctness requires jumping around in the codebase. We want to write code so that pieces of code which are dependent on eachother appear right next to eachother.

So the part where we grab `minFee` should be up at the top of the function, but then the part where we calculate the actual fee should be down by the transaction builder.

Reply:
Gotcha. Amended the last commit with the suggested changes.